### PR TITLE
Update rust, deps, fix exposed issues

### PR DESCRIPTION
**Pull request by dancrossnyc**
Update Rust and dependencies, and fix a number of exposed issues that popped up as a result.  This code is now clippy-clean.

### CODE REVIEW

**Review by orangecms — APPROVED** [review]
We had a similar upgrade in oreboot recently, looked all familiar. :)

File: x86_64/src/devcons.rs [hunk 1]
```diff
@@ -14,7 +15,7 @@ impl Uart for Uart16550 {
 
 pub fn init() {
     Console::new(|| {
-        static mut UART: Uart16550 = Uart16550 { port: 0x3f8 };
-        unsafe { &mut UART }
+        static UART: SyncUnsafeCell<Uart16550> = SyncUnsafeCell::new(Uart16550 { port: 0x3f8 });
```
Comment (gmacd) [inline x86_64/src/devcons.rs:18]: Should the same change be made to x86_64 devcons.rs?

**Review by gmacd — APPROVED** [review]

File: x86_64/src/devcons.rs (hunk 1 above)
Comment (dancrossnyc) [reply x86_64/src/devcons.rs:18]: I don't think we need to.

Comment (dancrossnyc) [reply x86_64/src/devcons.rs:18]: Oh wait, you mean aarch64? Yeah, probably.

Comment (gmacd) [reply x86_64/src/devcons.rs:18]: actually I meant x86-64 - I didn't realise this was already it 🤦

**Pull request merged**